src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,154 @@
+                    currentNode.next = new Node<>(hash(key), key, value, null);
+                    size++;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = table[getBucketNumber(key)];
+        while (currentNode != null) {
+            if (key == null && currentNode.key == null
+                    || Objects.equals(key, currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
+    public V remove(K key) {
+        int bucketNumberByKey = getBucketNumber(key);
+        Node<K, V> currentNode = table[bucketNumberByKey];
+        V removedValue = getValue(key);
+        if (currentNode.next == null) {
+            table[bucketNumberByKey] = null;
+        } else {
+            while (currentNode.next.key != key) {
+                currentNode = currentNode.next;
+            }
+            currentNode.next = currentNode.next.next;
+        }
+        size--;
+        return removedValue;
+    }
+
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public void print() {
+        int bucketNumber = 0;
+        for (Node<K, V> node : table) {
+            System.out.print(bucketNumber + " ");
+            if (node == null || node.next == null) {
+                System.out.println(node);
+            } else {
+                System.out.print(node);
+                while (node.next != null) {
+                    System.out.print(" --> " + node.next);
+                    node = node.next;
+                }
+                System.out.println();
+            }
+            bucketNumber++;
+        }
+    }
+
+    private int hash(Object key) {
+        return (key == null) ? 0 : key.hashCode();
+    }
+
+    private void resize() {
+        Node<K, V>[] oldTable = table;
+        int oldCapacity = oldTable.length;
+        int newCapacity = oldCapacity << 1;
+        threshold = threshold << 1;
+        table = (Node<K, V>[]) new Node[newCapacity];
+        size = 0;
+        for (Node<K, V> node : oldTable) {
+            Node<K, V> currentNode = node;
+            if (currentNode != null) {
+                if (currentNode.next == null) {
+                    put(currentNode.key, currentNode.value);
+                } else {
+                    while (currentNode != null) {
+                        put(currentNode.key, currentNode.value);
+                        currentNode = currentNode.next;
+                    }
+                }
+            }
+        }
+    }
+
+    private int getBucketNumber(K key) {
+        if (hash(key) < 0) {
+            return hash(key) % table.length * -1;
+        }
+        return hash(key) % table.length;
+    }
+
+    private static class Node<K, V> {
+        private int hash;
```

Comment:
Do you use this var?

Reply:
Only for "toString() --> print()" & "class Main" which has already been deleted :) SOrry again))